### PR TITLE
doge: move to pkgs/by-name & 3.5.0 -> 3.6.0 & add patch adding a small doge

### PR DESCRIPTION
**Pull request by quantenzitrone**
## Description of changes
Is it allowed to include a patch from an unmerged PR, or do i have to create a new package?
The [PR 66](https://github.com/Olivia5k/doge/pull/66) only adds a small doge, which is neat and should not introduce any bugs.
If not, I can remove the last commit, no problem.
~~If yes, is the version naming (`3.6.0+pr66`) ok?~~

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onemoresuza** [comment]
# Reviewed points

Result of `nixpkgs-review pr 267244` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>doge</li>
    <li>doge.dist</li>
  </ul>
</details>

- [X] package name fits guidelines
- [ ] package version fits guidelines
	- I've never seen a version like so: `${version}+prN` --- `rg --
	'version.*=.*\\+pr' pkgs/*` returns nothing.

	The [`vlc`][vlc] package, for example, fetches a patch, but does not mention
	it on its version.
- [X] package builds on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] all depending packages build
- [X] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [X] patches that are remotely available are fetched rather than vendored

[vlc]: <https://github.com/NixOS/nixpkgs/blob/1a0ebc9b6a392c30b2b0fbb007540bf556587382/pkgs/applications/video/vlc/default.nix#L216>

**Comment by quantenzitrone** [comment]
How about keeping the version as `3.6.0` and just mentioning adding the patch in the commit message?

**Comment by quantenzitrone** [comment]
on my last pr, i was asked to do the moving to pkgs/by-name in a separate commit

**Comment by onemoresuza** [comment]
> on my last pr, i was asked to do the moving to pkgs/by-name in a separate commit

I don't think that's needed as long as the commit message mentions the change to `pkgs/by-name`.

**Comment by quantenzitrone** [comment]
makes sense

**Review by onemoresuza — APPROVED** [review]

File: pkgs/by-name/do/doge/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Olivia5k";
+    repo = "doge";
+    rev = version;
+    hash = "sha256-72nRghD5k0ofrlvV3hEdyrr6uzV4+8At1bOCmRZTxhk=";
+  };
+
+  patches = [
+    # fetch patch from pull 66 which adds as small doge
```
Comment (pbsds) [inline pkgs/by-name/do/doge/package.nix:19]:
```suggestion
    # https://github.com/Olivia5k/doge/pull/66, adds a small doge
```

**Review by onemoresuza — APPROVED** [review]

File: pkgs/by-name/do/doge/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    owner = "Olivia5k";
+    repo = "doge";
+    rev = version;
+    hash = "sha256-72nRghD5k0ofrlvV3hEdyrr6uzV4+8At1bOCmRZTxhk=";
+  };
+
+  patches = [
+    # https://github.com/Olivia5k/doge/pull/66, adds a small doge
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/Olivia5k/doge/pull/66.patch";
```
Comment (AndersonTorres) [inline pkgs/by-name/do/doge/package.nix:21]: URLs with `pull` can be deleted easily. Use the URL below instead:

https://github.com/Olivia5k/doge/commit/14e3ccc0a3f1e91862492e20a34d008768a34039.patch

File: pkgs/by-name/do/doge/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      hash = "sha256-3F+7F9RuoiVWoN+69T7tM871AXX1IQbHqFxD+S3TjxQ=";
+    })
+  ];
+
+  meta = {
+    homepage = "https://github.com/Olivia5k/doge";
+    description = "wow very terminal doge";
```
Comment (AndersonTorres) [inline pkgs/by-name/do/doge/package.nix:28]:
```suggestion
    description = "Wow very terminal doge";
```

File: pkgs/by-name/do/doge/package.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  fetchpatch,
+}:
+python3Packages.buildPythonPackage rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/do/doge/package.nix:7]:
```suggestion
}:

python3Packages.buildPythonPackage rec {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Reorder the history.

1. The first commit for fileset migration:
```
doge: migrate to by-name hierarchy
```

2. The next for the version bump:
```
doge: 3.5.0 -> 3.6.0
```

3. The next one for the patch:
```
doge: patch adding a small doge
```

4. Shouldn't this patch be optional?

**Comment by quantenzitrone** [comment]
> Shouldn't this patch be optional?

This is what I asked initially, but since it only adds new functionality (small doge) without changing and affecting (except for the `--help` output) anything else, I think, that it doesn't necessarily need to be optional.

**Comment by AndersonTorres** [comment]
"Incredibly hideous formatting" is polemical, to say the least.
The inconsistency of spacing between the brackets and elements inside them can be easily regarded as "harmful to some fragile human eyes".

I will not pursue this nitpicks on code that is not mine. The interpreters are happy, after all!

**Review by AndersonTorres — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by AndersonTorres** [comment]
(removed by the author)

**Comment by pbsds** [comment]
Thanks for the contribution. :heart: 

I appreciate separating the by-name migration to a separate commit, because it makes review far easier. However I also prefer not to strain the patience of all the lovely contributors with small formatting nits.

**Pull request merged**